jdaviz/core/loaders/importers/spectrum2d/spectrum2d.py

```diff
@@ -131,13 +131,21 @@ def output(self):
         except Exception:
             data_unit = u.count
 
-        # FITS WCS is invalid, so ignore it.
-        if wcs.spectral.naxis == 0:
-            kw = {}
-        else:
-            kw = {'wcs': wcs}
-
-        return Spectrum1D(flux=data * data_unit, meta=metadata, **kw)
+        try:
+            return Spectrum1D(flux=data * data_unit, meta=metadata, wcs=wcs)
+        except ValueError:
+            # In some cases this will fail if there is no found spectral axis
+            # in the WCS, in that case, if the specutils parser would be successful,
+            # allows this to fail so that is used instead.  If that would also fail,
+            # then fallback on using no WCS
```

Comment:
I couldn't understand your comment on the first or third reads. Here's my attempt to reword, but I don't know 100% that I'm correct:
```suggestion
            # In some cases, the above call to Spectrum1D will fail if no 
            # spectral axis is found in the WCS. Even without a spectral axis, 
            # the Spectrum1D.read parser may work, so we try that next.
            # If that also fails, then drop the WCS.
```